rllib/offline/estimators/tests/test_ope.py

```diff
@@ -1,186 +1,204 @@
             gamma=self.gamma,
         )
         estimates = estimator.estimate(self.batch)
-        self.mean_ret[name] = estimates["v_target"]
-        self.std_ret[name] = estimates["v_target_std"]
+        assert estimates is not None, "WIS estimator did not compute estimates"
 
-    def test_dm_fqe(self):
-        name = "dm_fqe"
         estimator = DirectMethod(
             policy=self.algo.get_policy(),
             gamma=self.gamma,
-            q_model_config={"type": FQETorchModel, **self.q_model_config},
+            q_model_config=self.q_model_config,
         )
-        self.losses[name] = estimator.train(self.batch)
+        losses = estimator.train(self.batch)
+        assert losses, "DM estimator did not return mean loss"
         estimates = estimator.estimate(self.batch)
-        self.mean_ret[name] = estimates["v_target"]
-        self.std_ret[name] = estimates["v_target_std"]
+        assert estimates is not None, "DM estimator did not compute estimates"
 
-    def test_dr_fqe(self):
-        name = "dr_fqe"
         estimator = DoublyRobust(
             policy=self.algo.get_policy(),
             gamma=self.gamma,
-            q_model_config={"type": FQETorchModel, **self.q_model_config},
+            q_model_config=self.q_model_config,
         )
-        self.losses[name] = estimator.train(self.batch)
+        losses = estimator.train(self.batch)
+        assert losses, "DM estimator did not return mean loss"
         estimates = estimator.estimate(self.batch)
-        self.mean_ret[name] = estimates["v_target"]
-        self.std_ret[name] = estimates["v_target_std"]
+        assert estimates is not None, "DM estimator did not compute estimates"
 
     def test_ope_in_algo(self):
-        results = self.algo.evaluate()
-        print("OPE in Algorithm results")
-        estimates = results["evaluation"]["off_policy_estimator"]
-        mean_est = {k: v["v_target"] for k, v in estimates.items()}
-        std_est = {k: v["v_target_std"] for k, v in estimates.items()}
-
-        print("Mean:")
-        print(*list(mean_est.items()), sep="\n")
-        print("Stddev:")
-        print(*list(std_est.items()), sep="\n")
-        print("\n\n\n")
-
-    def test_fqe_model(self):
+        # Test OPE in DQN, during training as well as by calling evaluate()
+        results = self.algo.train()
+        ope_results = results["evaluation"]["off_policy_estimator"]
+        # Check that key exists AND is not {}
+        assert ope_results, "Did not run OPE in training!"
+        assert set(ope_results.keys()) == {
+            "is",
+            "wis",
+            "dm_fqe",
+            "dr_fqe",
+        }, "Missing keys in OPE result dict"
+
+        # Check algo.evaluate() manually as well
+        results = self.algo.train()
+        ope_results = results["evaluation"]["off_policy_estimator"]
+        assert ope_results, "Did not run OPE on call to Algorithm.evaluate()!"
+        assert set(ope_results.keys()) == {
```

Comment:
nit: it's better to use self.assertEqual() in places that you can like here. The error message will be more informative when the test fails.

Reply:
Take a look at other self.assertX functions. They can be used instead of you calling the assert with your custom messages.

Reply:
**TODO** please still use the set(ope_results.keys()) to form a set.